src/sentry/rules/conditions/event_frequency.py

```diff
@@ -396,27 +402,28 @@ def batch_query_hook(
         self, group_ids: set[int], start: datetime, end: datetime, environment_id: int
     ) -> dict[int, int]:
         batch_sums: dict[int, int] = defaultdict(int)
-        groups = Group.objects.filter(id__in=group_ids)
-        error_issues = [group for group in groups if group.issue_category == GroupCategory.ERROR]
-        generic_issues = [group for group in groups if group.issue_category != GroupCategory.ERROR]
+        groups = Group.objects.filter(id__in=group_ids).values_list("id", "type")
+        error_issue_ids = [group[0] for group in groups if group[1] == GroupCategory.ERROR.value]
+        generic_issue_ids = [group[0] for group in groups if group[1] != GroupCategory.ERROR.value]
```

Comment:
You'll need to copy this logic

https://github.com/getsentry/sentry/blob/3f6b07dbd53386c8b8bb44a84fbffcdd5d59f16f/src/sentry/models/group.py#L953-L959